File: examples/quick-test/nodejs/test_eval.js
/**
 * Quick test: Chivox MCP speech evaluation
 * Usage: node test_eval.js
 *
 * No external dependencies required (uses built-in fetch).
 */

const API_KEY = "sk-your-api-key";
const BASE_URL = "https://mcp.cloud.chivox.com";

let sessionId = null;

async function mcpRequest(method, params, reqId = 1) {
  const headers = {
    Authorization: `Bearer ${API_KEY}`,
    "Content-Type": "application/json",
  };
  if (sessionId) headers["Mcp-Session-Id"] = sessionId;

  const resp = await fetch(`${BASE_URL}/`, {
    method: "POST",
    headers,
    body: JSON.stringify({
      jsonrpc: "2.0",
      id: reqId,
      method,
      params,
    }),
  });

  if (!sessionId && resp.headers.get("Mcp-Session-Id")) {
    sessionId = resp.headers.get("Mcp-Session-Id");
  }

  return resp.json();
}

async function main() {
  // 1. Initialize
  console.log("=== Initialize ===");
  let result = await mcpRequest("initialize", {
    protocolVersion: "2024-11-05",
    capabilities: {},
    clientInfo: { name: "quick-test-js", version: "1.0.0" },
  });
  console.log(JSON.stringify(result, null, 2));

  // 2. List tools
  console.log("\n=== Tools List ===");
  result = await mcpRequest("tools/list", {}, 2);
  const tools = result?.result?.tools || [];
  tools.forEach((t) =>
    console.log(`  - ${t.name}: ${(t.description || "").slice(0, 60)}`)
  );

  // 3. Evaluate 'hello'
  console.log("\n=== Evaluate 'hello' ===");
  result = await mcpRequest(
    "tools/call",
    {
      name: "en_word_eval",
      arguments: {
        ref_text: "hello",
        audio_url: "https://dict.youdao.com/dictvoice?audio=hello&type=1",
      },
    },
    3
  );

  const content = result?.result?.content || [];
  if (content.length > 0 && content[0].type === "text") {
    console.log(JSON.stringify(JSON.parse(content[0].text), null, 2));
  } else {
    console.log(JSON.stringify(result, null, 2));
  }
}

main().catch(console.error);

File: examples/quick-test/python/test_eval.py
"""
Quick test: Chivox MCP speech evaluation
Usage: python test_eval.py

Prerequisites: pip install httpx
"""

import json
import httpx

API_KEY = "sk-your-api-key"
BASE_URL = "https://mcp.cloud.chivox.com"
HEADERS = {
    "Authorization": f"Bearer {API_KEY}",
    "Content-Type": "application/json",
}

session_id = None


def mcp_request(method, params, req_id=1):
    global session_id
    hdrs = {**HEADERS}
    if session_id:
        hdrs["Mcp-Session-Id"] = session_id

    resp = httpx.post(
        f"{BASE_URL}/",
        json={"jsonrpc": "2.0", "id": req_id, "method": method, "params": params},
        headers=hdrs,
        timeout=60,
    )

    if not session_id and "Mcp-Session-Id" in resp.headers:
        session_id = resp.headers["Mcp-Session-Id"]

    return resp.json()


# 1. Initialize
print("=== Initialize ===")
result = mcp_request("initialize", {
    "protocolVersion": "2024-11-05",
    "capabilities": {},
    "clientInfo": {"name": "quick-test", "version": "1.0.0"},
})
print(json.dumps(result, indent=2, ensure_ascii=False))

# 2. List tools
print("\n=== Tools List ===")
result = mcp_request("tools/list", {}, req_id=2)
for tool in result.get("result", {}).get("tools", []):
    print(f"  - {tool['name']}: {tool.get('description', '')[:60]}")

# 3. Call en_word_eval
print("\n=== Evaluate 'hello' ===")
result = mcp_request("tools/call", {
    "name": "en_word_eval",
    "arguments": {
        "ref_text": "hello",
        "audio_url": "https://dict.youdao.com/dictvoice?audio=hello&type=1",
    },
}, req_id=3)

content = result.get("result", {}).get("content", [])
if content and content[0].get("type") == "text":
    eval_result = json.loads(content[0]["text"])
    print(json.dumps(eval_result, indent=2, ensure_ascii=False))
else:
    print(json.dumps(result, indent=2, ensure_ascii=False))
